hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveShimLoader.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hive;
+
+import org.apache.hive.common.util.HiveVersionInfo;
+import org.apache.hudi.hive.client.HiveShim;
+import org.apache.hudi.hive.client.HiveShimV220;
+import org.apache.hudi.hive.client.HiveShimV210;
+import org.apache.hudi.hive.client.HiveShimV201;
+import org.apache.hudi.hive.client.HiveShimV122;
+import org.apache.hudi.hive.client.HiveShimV121;
+import org.apache.hudi.hive.client.HiveShimV111;
+import org.apache.hudi.hive.client.HiveShimV234;
+import org.apache.hudi.hive.client.HiveShimV230;
+import org.apache.hudi.hive.client.HiveShimV233;
+import org.apache.hudi.hive.client.HiveShimV100;
+import org.apache.hudi.hive.client.HiveShimV101;
+import org.apache.hudi.hive.client.HiveShimV110;
+import org.apache.hudi.hive.client.HiveShimV120;
+import org.apache.hudi.hive.client.HiveShimV200;
+import org.apache.hudi.hive.client.HiveShimV211;
+import org.apache.hudi.hive.client.HiveShimV231;
+import org.apache.hudi.hive.client.HiveShimV232;
+import org.apache.hudi.hive.client.HiveShimV235;
+import org.apache.hudi.hive.client.HiveShimV236;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A loader to load HiveShim.
+ */
+public class HiveShimLoader {
```

Comment:
if we copy the code from somewhere, we should include the project in LICENSE and NOTICE

Reply:
Will change.